src/Dapr.Actors/Runtime/Actor.cs

```diff
@@ -328,9 +291,39 @@ protected async Task UnregisterTimerAsync(IActorTimer timer)
         protected async Task UnregisterTimerAsync(string timerName)
         {
             await ActorRuntime.DaprInteractor.UnregisterTimerAsync(this.actorTypeName, this.Id.ToString(), timerName);
-            if (this.timers.ContainsKey(timerName))
+        }
+
+        private void ValidateTimerCallback(string callback)
+        {
+            var actorTypeName = this.Host.ActorTypeInfo.ActorTypeName;
+            var actorType = this.Host.ActorTypeInfo.ImplementationType;
+
+            // GetMethod will throw an AmbiguousMatchException if more than one methods are found with the same name
+            var methodInfo = actorType.GetMethod(callback, BindingFlags.Public | BindingFlags.Instance);
+
+            // Check if the method exists
+            if (methodInfo == null)
+            {
+                throw new ArgumentException($"Timer callback method: {callback} does not exist in the Actor class: {actorTypeName}");
+            }
+
+            // The timer callback can accept only 0 or 1 parameters
+            var parameters = methodInfo.GetParameters();
+            if (parameters.Length > 1)
+            {
+                throw new ArgumentException("Timer callback can accept only zero or one parameters");
+            }
+
+            // The timer callback should return only Task return type
+            if (methodInfo.ReturnType != typeof(Task))
+            {
+                throw new ArgumentException("Timer callback can only return type Task");
+            }
+
+            // The timer callback cannot be overridden in a derived class
+            if (methodInfo.IsVirtual)
```

Comment:
Why is that a problem?

Reply:
Removed this check. I don't think we need to check the abstract flag as well since we will invoke this method on an instance.